src/App.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { ChangeEvent, KeyboardEvent } from "react";
+
+interface CounterProps {
+  initialValue: number;
+}
+
+interface CounterState {
+  value: number;
+}
+
+class Counter extends React.Component<CounterProps> {
+  state: CounterState;
+
+  constructor(props: CounterProps) {
+    super(props);
+    this.state = {
+      value: props.initialValue,
+    };
+  }
+
+  increment = () => {
+    this.setState({ value: this.state.value + 1 });
+  };
```

Comment:
```suggestion
  increment() {
    this.setState({ value: this.state.value + 1 });
  };
```
Optional: if you have a bit of extra time try to define `increment` as function declaration. It will break because of how `this` works in JS. [In this doc](https://legacy.reactjs.org/docs/handling-events.html) you can find how to fix it. Hint: it's about `bind`ing.
This might be helpful if you work with a legacy codebase.

Reply:
I used arrow functions because I was getting that suggestion of binding, but now I read the docs and replaced all the handlers.